### PR TITLE
NXDRIVE-1098: Auto-Lock does not work when there are orphans

### PR DESCRIPTION
**Pull request by BoboTiG**
Also, a lot of cleanup:
- Framework: do not create test folder for case sensitivity in the Drive folder but in a temporary one
- Remove a debug line from InvalidBatchException
- `DirectEdit._url` -> `.url`
- Better logic when using DE if Drive is already running
- Review DirectEdit class
- Add utils.py::`simplify_url()` and refactor test_utils.py to use pytest format
- Removed `LocalClient.make_tree()`

### CODE REVIEW

File: nuxeo-drive-client/nxdrive/direct_edit.py [hunk 1]
```diff
@@ -424,41 +446,55 @@ def _handle_queues(self):
...
                 self._error_queue.push(ref, ref)
-                continue
+
+    def _handle_queues(self):
+        # Lock any document
+        self._handle_lock_queue()
+
+        # Unqueue any errors
+        item = self._error_queue.get()
+        while item:
```
Comment (LKleinNux) [inline nuxeo-drive-client/nxdrive/direct_edit.py]: Why not use a generator?

File: nuxeo-drive-client/nxdrive/direct_edit.py [hunk 2]
```diff
@@ -301,91 +317,96 @@ def edit(self, server_url, doc_id, user=None, download_url=None):
...
                 # Try again in 30s
                 log.exception('Cannot %s document %r', action, ref)
                 self.directEditLockError.emit(action, os.path.basename(ref), uid)
 
-        # Unqueue any errors
-        item = self._error_queue.get()
-        while item:
-            self._upload_queue.put(item.get())
-            item = self._error_queue.get()
+    def _hangle_upload_queue(self):
```
Comment (LKleinNux) [inline nuxeo-drive-client/nxdrive/direct_edit.py]: Typo here

File: nuxeo-drive-client/nxdrive/direct_edit.py [hunk 3]
```diff
@@ -485,70 +520,93 @@ def _stop_watchdog(self):
...
 
+    def _is_regular_file(self, name):
+        # type: (str) -> bool
+        """
+        Check if a given file name is not a temporary one created by
+        a tierce software.  We use it to disable the feature as we
+        use the global open files instead of editor lock file.
+        """
+
+        return False and self._is_lock_file(name)
```
Comment (LKleinNux) [inline nuxeo-drive-client/nxdrive/direct_edit.py]: Wouldn't that always return False?

**Comment by BoboTiG** [comment]
Changes done. Also:
- Review BlackListQueue class (and so tests)

**Pull request merged**